### ISSUE TITLE
Subclass cannot assign an attribute defined with a TypeVar with bound after 1.1.164

### ISSUE THREAD

**Issue by bellini666** [microsoft/pyright#2245]
Not sure if the description is the best one, but this piece of code should show the issue:

```python
from typing import Generic, Type, TypeVar


class A:
    pass


class B(A):
    pass


class MyStr(str):
    pass


_T = TypeVar("_T", bound=A)
_TT = TypeVar("_TT", bound=str)


class SomeClass(Generic[_T, _TT]):
    foo: Type[_T]
    bar: Type[_TT]
 
 
class SomeSubclass(SomeClass):
    foo = B
    bar = MyStr 


class OtherSubclass(SomeClass):
    foo = A
    bar = str
```

Note that in `SomeSubclass` I'm setting a subclass of the required types and in `OtherSubclass` I'm setting the same types that are bounded to their typevars. Running pyright 1.1.164 gives me this error:

```
❯ pyright test.py
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/test.py
  /tmp/test.py:26:11 - error: Expression of type "foo" cannot be assigned to declared type "Type[_T@SomeClass]"
    Type "foo" cannot be assigned to type "Type[_T@SomeClass]" (reportGeneralTypeIssues)
  /tmp/test.py:27:11 - error: Expression of type "bar" cannot be assigned to declared type "Type[_TT@SomeClass]"
    Type "bar" cannot be assigned to type "Type[_TT@SomeClass]" (reportGeneralTypeIssues)
  /tmp/test.py:31:11 - error: Expression of type "foo" cannot be assigned to declared type "Type[_T@SomeClass]"
    Type "foo" cannot be assigned to type "Type[_T@SomeClass]" (reportGeneralTypeIssues)
  /tmp/test.py:32:11 - error: Expression of type "bar" cannot be assigned to declared type "Type[_TT@SomeClass]"
    Type "bar" cannot be assigned to type "Type[_TT@SomeClass]" (reportGeneralTypeIssues)
4 errors, 0 warnings, 0 infos 
Completed in 0.597sec
```

This didn't happen in previous version.

**Comment by erictraut** [comment]
Thanks for reporting the problem. This is part of a new type check that I added in the last release. It contains a bug whereby it doesn't properly specialize the type of the base class variable. This will be fixed in the next release. Apologies for any inconvenience.

**Comment by erictraut** [comment]
I'll point out in your sample that the class declarations for `SomeSubclass` and `OtherSubclass` are not providing the proper type arguments for `SomeClass`, so a type checker will assume that they are `Any` in this case. That's probably not what you intend.

This would be preferable:
```python
class SomeSubclass(SomeClass[B, MyStr]):
    foo = B
    bar = MyStr

class OtherSubclass(SomeClass[A, str]):
    foo = A
    bar = str
```

If you enable the `reportMissingTypeArguments` diagnostic check, pyright will generate errors if you omit type arguments for generic classes.

**Comment by bellini666** [comment]
@erictraut oh yeah, I made the example in a hurry and forgot to add those. But the error happens even if you add those (but you probably already know that =P).

Thanks, and looking forward for this fix to be released! :)

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.165, which I just published. It will also be included in the next release of pylance.
